src/expandableCalendar/agendaListNew.tsx

```diff
@@ -0,0 +1,258 @@
+
+  const style = useRef(styleConstructor(theme));
+  const list = useRef<any>();
+  const _topSection = useRef(sections[0]?.title);
+  const didScroll = useRef(false);
+  const sectionScroll = useRef(false);
+  const sectionHeight = useRef(0);
+  const [data, setData] = useState([] as any[]);
+
+  useEffect(() => {
+    const items = sections.reduce((acc: any, cur: any) => {
+      return [...acc, {title: cur.title, isTitle: true}, ...cur.data];
+    }, []);
+    setData(items);
+
+    if (date !== _topSection.current) {
+      setTimeout(() => {
+        scrollToSection(date);
+      }, 500);
+    }
+  }, [sections]);
+
+  useDidUpdate(() => {
+    // NOTE: on first init data should set first section to the current date!!!
+    if (updateSource !== UpdateSources.LIST_DRAG && updateSource !== UpdateSources.CALENDAR_INIT) {
+      scrollToSection(date);
+    }
+  }, [date]);
+
+  const getSectionIndex = (date: string) => {
+    let dataIndex = 0;
+
+    for (let i = 0; i < sections.length; i++) {
+      if (sections[i].title === date) {
+        return dataIndex;
+      }
+      dataIndex += sections[i].data.length + 1;
+    }
+  };
+
+  const getNextSectionIndex = (date: string) => {
+    const cur = new XDate(date);
+    let dataIndex = 0;
+
+    for (let i = 0; i < sections.length; i++) {
+      const titleDate = parseDate(sections[i].title);
+      if (isGTE(titleDate,cur)) {
+        return dataIndex;
+      }
+      dataIndex += sections[i].data.length + 1;
+    }
+  };
+
+  const getSectionTitle = useCallback((title: string) => {
+    if (!title) return;
+
+    let sectionTitle = title;
+
+    if (dayFormatter) {
+      sectionTitle = dayFormatter(title);
+    } else if (dayFormat) {
+      if (useMoment) {
+        const moment = getMoment();
+        sectionTitle = moment(title).format(dayFormat);
+      } else {
+        sectionTitle = new XDate(title).toString(dayFormat);
+      }
+    }
+
+    if (markToday) {
+      const string = getDefaultLocale().today || todayString;
+      const today = isToday(title);
+      sectionTitle = today ? `${string}, ${sectionTitle}` : sectionTitle;
+    }
+
+    return sectionTitle;
+  }, []);
+
+  const scrollToSection = useCallback(debounce((d) => {
+    const sectionIndex = scrollToNextEvent ? getNextSectionIndex(d) : getSectionIndex(d);
+    if (isUndefined(sectionIndex)) {
+      return;
+    }
+
+    if (list?.current && sectionIndex !== undefined) {
+      sectionScroll.current = true; // to avoid setDate() in onViewableItemsChanged
+      _topSection.current = sections[findItemTitleIndex(sectionIndex)]?.title;
+
+      list.current?.scrollToIndex(sectionIndex, true);
+    }
+  }, 1000, {leading: false, trailing: true}), [ sections]);
+
+  const layoutProvider = useMemo(
+    () => new LayoutProvider(
+      (index) => data[index]?.isTitle ? 'title': 'page',
+      (type, dim) => {
+        dim.width = constants.screenWidth;
+        dim.height = type === 'title' ? 60 : 120;
```

Comment:
should see what the right size to pass

Reply:
@ethanshar @Inbal-Tish please give your thoughts

Reply:
In the current example, it should be 80 instead of 120, but anyways you'll have to let the user pass the item's height to set it there.

Reply:
Created `infiniteListProps` in the oldAgendaList, with an option to define these values (and load the new component in case it pass - even an empty object)

Reply:
@Inbal-Tish It's ok to put values like that? will it work in any view? tablet view? landscape?

Reply:
It will stay the same unless you change it when the orientation change. Regarding tablets, it depends on the design of course... Once you give the user the option to set these values it can be dynamic. 'pageWidth' and 'pageHeight' covers that